scripts/calendar/germany_school_generator.py

```
@@ -0,0 +1,543 @@
+        if end_year <= start_year:
+            end_year += 100
+    return start_year, end_year
+
+
+def _parse_pdf_table(path: Path) -> tuple[int, dict[str, list[tuple[str, str, str]]]]:
+    pdfplumber = _require_pdfplumber()
+    with pdfplumber.open(path) as pdf:
+        table = pdf.pages[0].extract_table()
+
+    if not table:
+        raise ValueError(f"Unable to extract table from {path.name}.")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)